config/initializers/diaspora_federation.rb

```diff
@@ -61,5 +35,26 @@
 
       person_entity.save!
     end
+
+    on :fetch_private_key_by_id do |diaspora_id|
+      Person.find_by(diaspora_handle: diaspora_id).owner.encryption_key
```

Comment:
We shouldn't need to allocate AR objects here, the right combination of `pluck` and `joins` should do it.